doc/avitype.1
.TH "avitype" "1" "12 February 2003" "Zdenek Kabelac" ""
.SH "NAME"
avitype \- show video info
.SH "SYNOPSIS"
.B avitype
.RI [options]\ [ \ files\  | \ URLs\  ]
.SH "DESCRIPTION"
.B avitype

.SH "OPTIONS"
\fB\-h \-\-help\fR
Show help message
.TP 

.BR 
.SH "SEE ALSO"
\fBaviplay(1)\fR


ChangeLog
2004/10/05
* ok the last release has one stupid bug in subtitle reader - fixed
  copying of subtitle lines where emptyline weren't copyied.
* as usually ffmpeg is regulary updated

2004/06/04
* yep - the time is running, everyone is using mplayer so who actully cares,
  but still it's worth to mention that some bug in threading was
  fixed with 2.6 kernels and aviplay usage
* Xft subtitles are now rendered with line breaking so if the line
  is too long, overlaping words are moved to the next line - this should
  be probably convigurable option we will see
* ffmpeg is contiuosly updated - but for now the framerate logic
  is changed way too offect so it doesn't currently work

2003/07/10
* nothing really inovative - mostly just ffmpeg gets update
  main purpose is to make a new Debian/SF release

2003/06/08 - alexannika@users.sourceforge.net
* new exp. multichannel view
* press [ or ] to change number of tiles
* press + or - to change speed of channel switch

2003/06/06 - alexannika@users.sourceforge.net
* added files samples/qtvidcap/frequencies.{c,h} (taken from xawtv)
* added frequencies.{c,h} to samples/qtvidcap/Makefile.am
* avicap now uses v4l->setFreq() to set frequency
* recording continues while you can zapp around

2003/06/05 - alexannika@users.sourceforge.net
* sdl-renderer doesn't crash anymore
* menu & some keys work in sdl window
* you can switch between overlay/preview/sdl

2003/06/02 - alexannika@users.sourceforge.net
* avirec shows correct time

2003/05/30 - alexannika@users.sourceforge.net
* minor fixes

2003/05/26 - alexannika@users.sourceforge.net
* removed my savename code from
  lib/codeckeeper.cpp
* disabled the savename code in vidconfig.cpp/capproc.cpp
  the code is still there, but no named codecs are used
* no named codecs till I have the new version written
* removed the bogus constructor from lib/aviplay/aviplay_impl.h
* improved a little bit the sdl renderer
* fixed the avirec problem
* compiles with qt2.3.2 and without qt

2003/05/25
* still more cleanup & speedup for automake
* more headers moved to avm_ prefix usage
* headers are now stored within avifile0.7

2003/05/17
* again played a bit with configure script - cleanup
  support for divxbin has been added

2003/05/11
* cleaned some parts of avicap
* partly enable ffmpeg encoding support - works but not yet properly
* wow - we could really play SVQ3 via ffmpeg
* AAC should be working as well
  >> name codecs should be replaced with something more universal

2003/05/10 - alexannika@users.sourceforge.net
* fs-fenderer: yv12 gets negative picture, yuy2 works
* funny channel dependent cpu-usage

2003/05/09 - alexannika@users.sourceforge.net
* experimental! fullscreen/sdl/avicap-renderer
* added files samples/qtvidcap/fullscreen_renderer.{cpp,h}

2003/05/07 - alexannika@users.sourceforge.net
* user can now use 'named codecs'

2003/05/01 - alexannika@users.sourceforge.net
* avicap_de.qm gets installed to $(datadir)/avifile/i18n
* documentation gets installed to $(datadir)/avifile/doc
* added AUTHORS file to topdir
* README INSTALL COPYING AUTHORS ChangeLog gets installed to
  $(datadir)/avifile
* added global.h,huffman.h,layer12.h,layer3.h,version.c
  to plugins/libmad/libmad/Makefile.am (tarball didn't compile)

2003/05/01 - alexannika@users.sourceforge.net
* german translation of avicap done by Jochen Puchalla

2003/04/30 - alexannika@users.sourceforge.net
* more strings to translate
* slightly improved epg drawing

2003/04/30 - alexannika@users.sourceforge.net
* segmenting evaluates the dirpool
* file gets segmented into another dir when keepfree>freespace
* audio/video error & timeshift get shown in avicapwnd
* updated documentation

2003/04/29 - alexannika@users.sourceforge.net
* prepared for translation
* added samples/qtvidcap{avicap.pro,avicap_de.ts,avicap_de.qm}
* keyframe bug fixed
  *thanks* to Jochen Puchalla again :-)

2003/04/27 - alexannika@users.sourceforge.net
* avicap does synchfixes again
  *thanks* to Jochen Puchalla for his help & inspiration

2003/04/25 - alexannika@users.sourceforge.net
* slight improvements to the timertable

2003/04/22 - alexannika@users.sourceforge.net
* paranoid logging with -DAVICAP_PARANOID
* user logging to ~/.avm/avicap-userlog.log

2003/04/20
* implemneted support for I410 (used by ffmpeg indeo decoder)
* fixed Raw Uncompressed Bitmap support (it's stored with even columns
  for odd widths)

2003/04/16 - alexannika@users.sourceforge.net
* timertable: 'every' recordings get now updated automatically
* you can't switch channel to blocked programs
* new config tab 'other'
* other tab: password/child lock
* other tab: log to file checkbox (not yet functional)

2003/04/15 - alexannika@users.sourceforge.net
* epg faster redraw
* guide can now show range upto 3 days
* preliminary zoom (not yet working correctly)

2003/04/15
* nothing really impressive - just syncing with ffmpeg and cleaning few
  minor issues

2003/04/14 - alexannika@users.sourceforge.net
* epg browser highlights program which are running 'now'
* epg browser get updated from timertable when next/current
  recording has changed or any recording modified
* epg browser gets redrawn every minute (it's slow yet)

2003/04/13 - alexannika@users.sourceforge.net
* regexps get now written to ~/.avm/avicap-regexps
* epg-channel-id's for all providers the same
* you can change provider (but have to restart avicap yet)
* epgwindow interacts with timertable, you can put recordings from
  epgwindow into the timertable
* recordings from the timertable will show up shaded in the epg browser
* press middle button & move mouse to move display around
* regexp-edit separate window
* updated epg documentation
* epg browser compiles with Qt2.3.2 now

2003/04/12 - alexannika@users.sourceforge.net
* epgwindow regexps now much faster and usuable

2003/04/11 - alexannika@users.sourceforge.net
* epgwindow highlights program under mouse correctly
* epgwindow highlight/blocks programs by regexp
* fixes for Qt2.3.2, no epg yet


samples/misc/avitype.cpp
//                              -*- Mode: C++ -*-
// avitype.cpp -- Print info about a .AVI file
//
// $Id: avitype.cpp,v 1.12 2003/05/26 08:57:33 kabi Exp $
//
// Copyright (C) 2001  Tom Pavel <pavel@alum.mit.edu>
//
// Creator         : Tom Pavel  <tom>     Sat Feb  3 01:27:20 2001
// Modifier        : Tom Pavel  <tom>     Sun May 13 00:56:10 2001
// Update Count    : 34
// Status          : Unknown, Use with caution!
//



#include <avifile.h>
#include <aviplay.h>
#include <avm_fourcc.h>
#include <avm_except.h>
#include <utils.h>
#include <version.h>

#include <unistd.h>		// for getopt()
#include <stdio.h>
#include <stdlib.h>

static void Usage(const char* progname)
{
    fprintf(stderr, "Usage: %s  [-h] file1 file2 ...\n",  progname);
    exit(1);
}

static void printFile(const char* fname)
{
    avm::IReadFile* aviFile = avm::CreateReadFile(fname);
    if (aviFile == 0)
	return;
#if 0
    MainAVIHeader hdr;
    if (aviFile->GetHeader(&hdr, sizeof(hdr)) != 0)
	return;

    cout << endl;
    cout << "AVI file:    " << fname << endl;
    cout << "    Frames:  " << hdr.dwTotalFrames
	 << " of " << hdr.dwWidth << "x" << hdr.dwHeight << endl;
    cout << "    Rate:    " << setprecision(4)
	 << 1e6/(double)hdr.dwMicroSecPerFrame << " frame/sec"
	 << " or " << hdr.dwMaxBytesPerSec/1000.0 << " kB/sec" << endl;
    cout << "    Streams: " << hdr.dwStreams << " ("
	 << aviFile->VideoStreamCount() << "V + "
	 << aviFile->AudioStreamCount() << "A)"
	 << endl;

    IAviReadStream* vidStream = aviFile->GetStream (0, AviStream::Video);
    if (vidStream == 0) {
	cout << "Bad Video Stream" << endl;
	return;
    }
    AVIStreamHeader header;
    vidStream->GetHeader(&header, sizeof(header));
    cout << "    Video:   " << avm_fccToString(header.fccType)
	<< " " << avm_fcc_name(header.fccHandler)
	<< endl;

    IAviReadStream* audStream = aviFile->GetStream (0, AviStream::Audio);
    if (audStream == 0) {
	cout << "Bad Audio Stream" << endl;
	return;
    }
    WAVEFORMATEX wvFmt;
    if (audStream->GetAudioFormatInfo(&wvFmt, 0) != 0) {
	cout << "Failed to decode Audio Format" << endl;
	return;
    }
    cout << "    Audio:   " << avm_wave_format_name(wvFmt.wFormatTag)
	 << " " << wvFmt.wBitsPerSample << "-bit "
	 << wvFmt.nChannels << "-chan "
	 << wvFmt.nSamplesPerSec << " Hz" << endl;
#endif

}

int main(int argc, char* argv[])
try
{
    int debug = 0;

    int ch;
    while ((ch = getopt(argc, argv, "dhl:")) != EOF) {
	switch ((char)ch) {
	case 'd':
	    ++debug;
	    break;

	case 'h':
	case '?':
	default:
	    Usage (argv[0]);
	}
    }

    argc -= optind;
    argv += optind;
    if (argc < 1) Usage (argv[0]);


    // Standard AVIlib sanity check:
    if ( GetAvifileVersion() != AVIFILE_VERSION) {
	printf("This binary was compiled for Avifile ver. %x"
	       ", but the library is ver. %x. Aborting.\n",
	       AVIFILE_VERSION, GetAvifileVersion());
	return 0;
    }

    while (argc > 0) {
	printFile (*argv);
	--argc;
	++argv;
    }

}
catch(FatalError& error) {
    error.Print();
}

